concrete-core/src/specification/engines/glwe_ciphertext_tensor_product_computation.rs

```diff
@@ -0,0 +1,55 @@
+use super::engine_error;
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::{
+    GlweCiphertextEntity,
+};
+use crate::backends::core::private::crypto::glwe::GlweCiphertext;
+use crate::prelude::GlweCiphertext32;
+
+engine_error! {
+    GlweCiphertextTensorProductError for GlweCiphertextTensorProductEngine @
+    PolynomialSizeMismatch => "The polynomial size of the input and output GLWE ciphertexts must be\
+     the same.",
+    InputGlweDimensionMismatch => "The GLWE dimension of the input ciphertexts must be the same.",
+    OutputGlweDimensionMismatch => "The GLWE dimension of the output ciphertext is incorrect"
```

Comment:
@agnesLeroy this (currently) isn't used -- I'm unsure if we can check function outputs using `engine_error`. Do you know if it is possible?

Reply:
The errors you're having come from a mismatch in the error names in the generic check I think. :smile: we can look at it tomorrow :wink:

Reply:
To check the output dimension you need to verify it matches the formula (k(1 + (k+1)/2 ), right?

Reply:
Remove `OutputGlweDimensionMismatch`